dataflux_pytorch/integration_tests/integration_test.py

```diff
@@ -0,0 +1,193 @@
+"""
+ Copyright 2024 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ """
+
+import unittest
+import time
+import os
+import numpy
+import io
+import argparse
+
+from math import ceil
+from torch.utils import data
+from dataflux_pytorch import dataflux_iterable_dataset, dataflux_mapstyle_dataset
+
+
+class IntegrationTest(unittest.TestCase):
+    def get_config(self):
+        config = {}
+        # Gather env vars into dictionary.
+        config["project"] = os.getenv("PROJECT")
+        config["bucket"] = os.getenv("BUCKET")
+        config["prefix"] = os.getenv("PREFIX")
+        config["num_workers"] = os.getenv("LIST_WORKERS")
+        config["expected_file_count"] = os.getenv("FILE_COUNT")
+        config["expected_total_size"] = os.getenv("TOTAL_FILE_SIZE")
+        config["max_compose_bytes"] = os.getenv("MAX_COMPOSE_BYTES")
+        config["list_timeout"] = os.getenv("LIST_TIMEOUT")
+        config["download_timeout"] = os.getenv("DOWNLOAD_TIMEOUT")
+        config["parallelization"] = os.getenv("PARALLELIZATION")
+
+        # Type convert env vars.
+        if config["num_workers"]:
+            config["num_workers"] = int(config["num_workers"])
+        if config["expected_file_count"]:
+            config["expected_file_count"] = int(config["expected_file_count"])
+        if config["expected_total_size"]:
+            config["expected_total_size"] = int(config["expected_total_size"])
+        config["max_compose_bytes"] = (
+            int(config["max_compose_bytes"])
+            if config["max_compose_bytes"]
+            else 100000000
+        )
+        if config["list_timeout"]:
+            config["list_timeout"] = float(config["list_timeout"])
+        if config["download_timeout"]:
+            config["download_timeout"] = float(config["download_timeout"])
+        config["parallelization"] = (
+            int(config["parallelization"]) if config["parallelization"] else 1
+        )
+
+        return config
+
+    def test_list_and_load_iter(self):
+        list_start_time = time.time()
```

Comment:
A good chunk of the code here, such as defining the `data_format_fn` and the for-loop iteration that simulates the training are having some common code. Can we refactor those into some common helper code?